cmd/launcher/updater/updater.go

```diff
@@ -0,0 +1,153 @@
+package updater
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/actor"
+	"github.com/kolide/launcher/pkg/autoupdate"
+	"github.com/kolide/updater/tuf"
+)
+
+// UpdaterConfig is a struct of update related options. It's used to
+// simplify the call to `createUpdater` from launcher's main blocks.
+type UpdaterConfig struct {
+	Logger             log.Logger
+	RootDirectory      string // launcher's root dir. use for holding tuf staging and updates
+	AutoupdateInterval time.Duration
+	UpdateChannel      autoupdate.UpdateChannel
+	InitialDelay       time.Duration // start delay, to avoid whomping critical early data
+	NotaryURL          string
+	MirrorURL          string
+	NotaryPrefix       string
+	HTTPClient         *http.Client
+	SigChannel         chan os.Signal
+}
+
+// updater allows us to mock *autoupdate.Updater during testing
+type updater interface {
+	Run(opts ...tuf.Option) (stop func(), err error)
+}
+
+type updaterCmd struct {
+	updater                 updater
+	ctx                     context.Context
+	stopChan                chan bool
+	stopExecution           func()
+	config                  *UpdaterConfig
+	runUpdaterRetryInterval time.Duration
+}
+
+func (u *updaterCmd) execute() error {
+	// Failing to start the updater is not a fatal launcher
+	// error. If there's a problem, sleep and try
+	// again. Implementing this is a bit gnarly. In the event of a
+	// success, we get a nil error, and a stop function. But I don't
+	// see a simple way to ensure the updater is still running in
+	// the background.
+
+	// When launcher first starts, we'd like the
+	// server to start receiving data
```

Comment:
nit -- reflow this

Reply:
I think the working is good, just moved the comments closer to the code they were commenting